src/sage/libs/flint/qsieve.pyx

```diff
@@ -0,0 +1,69 @@
+"""
+Interface to FLINT's ``qsieve_factor()``. This used to interact
+with an external "QuadraticSieve" program, but its functionality has
+been absorbed into flint.
+"""
+
+from cysignals.signals cimport sig_check, sig_on, sig_off
+
+from sage.libs.flint.fmpz cimport fmpz_t, fmpz_init, fmpz_set_mpz, fmpz_get_mpz
+from sage.libs.flint.fmpz_factor cimport *
+from sage.libs.flint.qsieve cimport *
+from sage.libs.gmp.mpz cimport mpz_t, mpz_init, mpz_set, mpz_clear
+from sage.rings.integer cimport Integer
+
+def qsieve(n):
+    r"""
+    Factor ``n`` using the quadratic sieve.
+
+    INPUT:
+
+    - ``n`` -- an integer; neither prime nor a perfect power.
+
+    OUTPUT:
+
+    A list of the factors of ``n``. There is no guarantee that the
+    factors found will be prime, or distinct.
+
+    EXAMPLES::
+
+        sage: k = 19; n = next_prime(10^k)*next_prime(10^(k+1))
+        sage: factor(n)  # (currently) uses PARI
+        10000000000000000051 * 100000000000000000039
+        sage: qsieve(n)
+        [(10000000000000000051, 1), (100000000000000000039, 1)]
+
+    """
+    if n.is_zero():
```

Comment:
In this line you already assume that `n` is of type `Integer`?

Reply:
There were a few mistakes here. I've updated that function to call `n = Integer(n)` first, and then afterwards, it raises an `ArithmeticError` if `n.is_zero()`. That's what the other user-facing factorization functions do.